apps/nextjs-website/src/lib/strapi/tutorial.ts

```diff
@@ -21,6 +21,7 @@ const BaseTutorialAttributesCodec = t.strict({
   title: t.string,
   slug: t.string,
   image: t.strict({ data: t.union([NullToUndefinedCodec, MediaCodec]) }),
+  product: t.strict({ data: ProductCodec }),
```

Comment:
Maybe here can be used BaseProduct
```suggestion
  product: t.strict({ data: BaseProductCodec }),
```